apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/DatasetsResourceTest.java

```diff
@@ -1672,6 +1682,73 @@ void getDatasets__whenFetchingAllDatasets__thenReturnDatasetsSortedByCreatedDate
             findAndAssertPage(actualEntity, expected.size(), expected.size(), 1, expected.reversed());
         }
 
+        @ParameterizedTest
+        @MethodSource("sortDirectionProvider")
+        @DisplayName("when fetching all datasets, then return datasets sorted by last_created_experiment_at")
+        void getDatasets__whenFetchingAllDatasets__thenReturnDatasetsSortedByLastCreatedExperimentAt(Direction requestDirection, Direction expectedDirection) {
+            String workspaceName = UUID.randomUUID().toString();
+            String apiKey = UUID.randomUUID().toString();
+            String workspaceId = UUID.randomUUID().toString();
+
+            mockTargetWorkspace(apiKey, workspaceName, workspaceId);
+
+            List<Dataset> expected = PodamFactoryUtils.manufacturePojoList(factory, Dataset.class);
+            Set<DatasetLastExperimentCreated> datasetsLastExperimentCreated = new HashSet<>();
+
+            expected.forEach(dataset -> {
+                var id = createAndAssert(dataset, apiKey, workspaceName);
+                datasetsLastExperimentCreated.add(new DatasetLastExperimentCreated(id, Instant.now()));
+            });
+
+            mySqlTemplate.inTransaction(WRITE, handle -> {
+
+                var dao = handle.attach(DatasetDAO.class);
+                dao.recordExperiments(workspaceId, datasetsLastExperimentCreated);
+
+                return null;
```

Comment:
I believe there is no need for this, you can just user the order experiments are created to know the expected order

Reply:
Ok, can we just extract this to another method?

Reply:
Also, another option is to move this test to: `DatasetExperimentE2ETest`